pkg/controller/jobframework/base_webhook_test.go

```diff
@@ -438,3 +431,42 @@ func TestValidateOnUpdate(t *testing.T) {
 		})
 	}
 }
+
+func validateUpdateFromObject(oldGj, newGj jobframework.GenericJob) func(runtime.Object) jobframework.GenericJob {
+	return func(object runtime.Object) jobframework.GenericJob {
+		job, ok := object.(*batchv1.Job)
+		if !ok || job == nil {
+			return nil
+		}
+		if job.ResourceVersion == "1" {
+			return oldGj
+		}
+		return newGj
+	}
+}
+
+type mockGenericJobWithManagedBy struct {
+	*mocks.MockGenericJob
+	*mocks.MockJobWithManagedBy
+}
+
+func newMockGenericJobWithManagedBy(ctrl *gomock.Controller) *mockGenericJobWithManagedBy {
```

Comment:
This is probably OK. That said, I try to avoid shared boilerplate in tests. Shared helper functions are great, shared types less so. The problem with shared types is they often leak extra setup into cases that do not need it and obscure test intent.

Concrete example, `MockJobWithManagedBy: mocks.NewMockJobWithManagedBy(ctrl),` is not needed for `ValidateOnUpdate*`, yet it is pulled in because the shared type also serves `ValidateOnCreate`.